packages/tpetra/core/src/Tpetra_Details_makeColMap_def.hpp

```diff
@@ -532,7 +532,7 @@ struct ListGIDs
       }
       lcount++;
     }
-    if(i == (GO) present.size() - 1 && finalPass)
+    if((i == (GO) present.size() - 1) && finalPass)
```

Comment:
Nice catch!  Also, prefer C++ casts to C-style casts.  I would recommend the following, since it will work even for those rare users that have enabled unsigned GO:
```c++
if((i == static_cast<GO>(present.size() - 1)) && finalPass)
```